src/stim/simulators/vector_simulator.inl

```diff
@@ -21,14 +21,15 @@
 #include "stim/probability_util.h"
 #include "stim/stabilizers/pauli_string.h"
 
-using namespace stim;
+namespace stim {
 
-VectorSimulator::VectorSimulator(size_t num_qubits) {
+template <size_t W>
+VectorSimulator<W>::VectorSimulator(size_t num_qubits) {
```

Comment:
Wait, why is the vector simulator getting a word width? It doesn't store any simd bit fields. Shouldn't only some of its methods require this parameter?

Reply:
Probably an oversight on my part.

You're right, it has no member data that depends on W. I must have templatised it because of the use of the templatised class `PauliStringRef`.

I'll change this so that templatisation only touches the 4 member functions in the VectorSimulator and not the class as a whole.

Generally this was the pattern I have been trying to following - templatise functions unless member data also needs the template parameter.

Do you foresee any utility in having the class template here in future or shall I make this change now?

Reply:
Make the change back to not-templated-class. VectorSimulator works with floats not bits so it's not relevant to it.

Reply:
Change made.